### PR TITLE
[dagster-airlift][tutorial] Assets forward tutorial

### PR DESCRIPTION
**Pull request by dpeng817**
## Summary & Motivation
Assets forward version of the tutorial. Also changes some of the wording
## How I Tested These Changes
Existing tests
## Changelog
NOCHANGELOG

### CODE REVIEW

**Comment by dpeng817** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25193?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#25193** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25193?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#25168** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25168?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25160** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25160?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25159** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25159?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25158** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25158?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25139** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25139?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25138** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25138?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25127** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25127?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#25128](https://github.com/dagster-io/dagster/pull/25128) <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25128?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#25126** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25126?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25125** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25125?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25124** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25124?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25120** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25120?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dpeng817 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: examples/experimental/dagster-airlift/examples/tutorial-example/README.md [hunk 1]
```diff
@@ -476,59 +471,61 @@ def airflow_dags_path() -> Path:
...
-                    duckdb_database_name="jaffle_shop",
-                ),
-            ),
-        ),
-        task_defs(
-            "build_dbt_models",
+    return _multi_asset
+
+
+def rebuild_customer_list_assets() -> Sequence[Union[AssetsDefinition, AssetSpec]]:
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/examples/tutorial-example/README.md:490]: Union type hints are rough. Not sure we have better alternative.

**Review by schrockn — CHANGES_REQUESTED** [review]
Unless necessary for tests, I don't think we should have `rebuild_customer_list_assets` as separate functions. Would rather have those inlined.

**Comment by dpeng817** [comment]
Yea, I don't think it's necessary anymore since we only have one top-level definitions object now. Before it would induce multiple to inline them.

**Review by schrockn — APPROVED** [review]

File: examples/experimental/dagster-airlift/examples/tutorial-example/README.md [hunk 2]
```diff
@@ -211,35 +209,22 @@ def dbt_project_path() -> Path:
...
-    )
+@dbt_assets(
+    manifest=dbt_project_path() / "target" / "manifest.json",
+    project=DbtProject(dbt_project_path()),
+)
+def dbt_project_assets(context: AssetExecutionContext, dbt: DbtCliResource):
+    yield from dbt.cli(["build"], context=context).stream()
+
+
+mapped_assets = assets_with_task_mappings(
```
[the remark this answers is not in the document]
Comment (dpeng817) [reply examples/experimental/dagster-airlift/examples/tutorial-example/README.md:220]: I think it makes things a bit more clear. I tend to favor smaller variable declarations over one large nested declaration for tutorial code.

[the remark this answers is not in the document]
Comment (schrockn) [reply examples/experimental/dagster-airlift/examples/tutorial-example/README.md:220]: ok np. I don't feel strongly either way!

**Pull request merged**